internal/providers/terraform/azure/app_service_environment.go

```diff
@@ -0,0 +1,101 @@
+package azure
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/infracost/infracost/internal/schema"
+
+	"github.com/shopspring/decimal"
+)
+
+func GetAzureRMAppIsolatedServicePlanRegistryItem() *schema.RegistryItem {
+	return &schema.RegistryItem{
+		Name:  "azurerm_app_service_environment",
+		RFunc: NewAzureRMAppIsolatedServicePlan,
+		ReferenceAttributes: []string{
+			"resource_group_name",
+		},
+		Notes: []string{
+			"Costs associated with running an app service plan in Azure",
+		},
+	}
+}
+
+func NewAzureRMAppIsolatedServicePlan(d *schema.ResourceData, u *schema.UsageData) *schema.Resource {
+	return &schema.Resource{
+		Name:           d.Address,
+		CostComponents: AppIsolatedServicePlanCostComponent(d, u),
+	}
+}
+func AppIsolatedServicePlanCostComponent(d *schema.ResourceData, u *schema.UsageData) []*schema.CostComponent {
+
+	tier := d.Get("pricing_tier").String()
+	group := d.References("resource_group_name")
+	location := group[0].Get("location").String()
```

Comment:
@aliscott [location](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/app_service_environment#location) is mentioned as an Attribute for this resource in Terraform, does that mean this code can just do `d.Get("location").String()` or does it have to get it from `resource_group_name` if it's set and otherwise get it from the `subnet_id` (since that's required).

Reply:
I think it we will have to get it from `resource_group_name` or the `azurerm_virtual_network` of the subnet if we can't find it in the resource group, since attributes are only known after apply.